rskj-core/src/main/java/co/rsk/trie/Trie.java

```diff
@@ -1370,4 +1370,56 @@ private static TrieKeySlice deserialize(ByteBuffer message, boolean sharedPrefix
             return TrieKeySlice.fromEncoded(encodedKey, 0, lshared, lencoded);
         }
     }
+
+    // Additional auxiliary methods for Merkle Proof
+
+    @Nullable
+    public List<Trie> getNodes(byte[] key) {
```

Comment:
I think this whole list can be acquired through the use of the currently existing _iterators_. If that's the case, I would prefer to not add these methods to this class

Reply:
On a second thought, it's not easy to implement using the existing _iterators_